test/library/packages/UnitTest/Launcher.chpl

```diff
@@ -0,0 +1,164 @@
+    //std output pipe
+    while exec.stdout.readline(line) {
+      if line.strip() == separator1 then sep1Found = true;
+      else if line.strip() == separator2 && sep1Found {
+        select flavour {
+          when "ERROR" do result.addError(testName, tempString);
+          when "FAIL" do result.addFailure(testName, tempString);
+          when "SKIPPED" do result.addSkip(testName,tempString);
+        }
+        tempString = "";
+        sep1Found = false;
+      }
+      else if sep1Found then tempString+=line;
+      else {
+        var temp = line.strip().split(":");
+        if temp[1].strip().endsWith("]") {
+          var strSplit = temp[1].strip().split("]");
+          var testNameIndex = strSplit[1].split("[");
+          testName = fileName+": "+testNameIndex[1];
+          curIndex = testNameIndex[2]: int;
+          result.startTest();
+          if temp.size > 1 {
+            flavour = temp[2].strip();
+            if flavour == "OK" then result.addSuccess(testName);
+          }
+          tempString = "";
+        }  
+      }
+    }
+    //this is to check the error
+    if exec.stderr.readline(line) { 
+      tempString = line;
+      while exec.stderr.readline(line) do tempString+=line;
+      if testName!="" {
+        result.addError(testName,tempString);
+        haltOccured =  true;
+      }
+    }
+    exec.wait();//wait till the subprocess is complete
+    if haltOccured then runAndLog(executable, fileName, result, curIndex);
+  }
+
+  pragma "no doc"
+  /*Docs: Todo*/
+  proc testFile(file, ref result) throws {
+    var fileName = basename(file);
+    if fileName.endsWith(".chpl") && fileName.startsWith("test") {
+      var line: string;
+      var compErr = false;
+      var tempName = fileName.split(".chpl");
+      var executable = tempName[1];
+      if isFile(executable) {
+        var execRem = spawn(["rm",executable]);
+      }
+      var sub = spawn(["chpl",file,"-o",executable,"-M."],stderr = PIPE); //Compiling the file
+      if sub.stderr.readline(line) {
+        writeln(line);
+        compErr = true;
+      }
+      sub.wait();
+      if !compErr {
+        runAndLog(executable,fileName,result);
+        if !keepExec:bool {
+          var execRem = spawn(["rm",executable]);
+        }
+      }
+      else {
+        writeln("Compilation Error in ",fileName);
+        writeln("Possible Reasons can be passing a non-test function to UnitTest.runTest()");
+      }
+    }
+  }
+
+  pragma "no doc"
+  /*Docs: Todo*/
+  proc testDirectory(dir, ref result) throws {
+    for file in findfiles(startdir = dir, recursive = subdir:bool) {
+      testFile(file, result);
+    }
+  }
+
+  proc main(args: [] string) {
+    var dirs: [1..0] string,
+        files: [1..0] string;
+    var hadInvalidFile = false;
+
+    for a in args[1..(args.size-1)] {
+      try! {
+        if isFile(a) {
+          files.push_back(a);
+        }
+        else if isDir(a) {
+          dirs.push_back(a);
+        }
+        else {
```

Comment:
How about we check for `-h` or `--help` and emit a simple message for now too?

We can polish the help message contents later.

Reply:
Sure, I will add the -h option.